content/blog/deploying-mysql-schemas-using-dynamic-providers/index.md

```diff
@@ -0,0 +1,284 @@
+---
+title: "Deploying a MySQL schema using Dynamic Providers"
+date: 2020-08-16T15:06:26-07:00
+meta_desc: Leveraging Pulumi Dynamic Providers to expand opportunities in cloud architecture design
+meta_image: meta.png
+authors: ["vova-ivanov"]
+tags: ["aws", "python", "mysql"]
+---
+
+Previously, we created a Python voting application using Flask and Redis. In this blog post, we will explore how to create a MySQL database and initialize it with a schema and some data. What seems to be a simple step is much more interesting than it appears, however, because Pulumi does not have a built-in library for such a task. To do it, we will create our own library: a Dynamic Provider.
```

Comment:
"In this blog post, we will explore how to create a MySQL database and initialize it with a schema and some data"

We have a mysql provider, are you sure this is what you mean?

Reply:
The existing Pulumi MySQL provider (https://www.pulumi.com/docs/reference/pkg/mysql/) allows me to create everything except tables, and I do use it in the blog post code. However, I want to create tables as part of the infrastructure itself, since  tables themselves can be considered a type of infrastructure. Because creating tables requires admin credentials, doing it this way, allows me to limit the sharing of admin permissions only to `pulumi up`, so the application itself runs under restricted permissions at all times.